### PR TITLE
feat(emqx_bridge): add /bridges_probe API endpoint

### PR DESCRIPTION
**Pull request by sstrigler**
Fixes [EMQX-8571](https://emqx.atlassian.net/browse/EMQX-8571)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/emqx/emqx/blob/master/CONTRIBUTING.md.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [x] Changed lines covered in coverage report
- [x] Change log has been added to `changes/` dir
- [x] For internal contributor: there is a jira ticket to track this change
- [x] Coordinate with Dashboard-Team

### CODE REVIEW

File: apps/emqx_bridge/src/emqx_bridge_api.erl [hunk 1]
```diff
@@ -68,7 +69,8 @@ paths() ->
         "/bridges/:id",
         "/bridges/:id/operation/:operation",
         "/nodes/:node/bridges/:id/operation/:operation",
-        "/bridges/:id/reset_metrics"
+        "/bridges/:id/reset_metrics",
+        "/bridges_test"
```
Comment (zmstone) [inline apps/emqx_bridge/src/emqx_bridge_api.erl:73]: try not to use _test suffix (clashes test code).
maybe bridges_probe.

File: apps/emqx_bridge/i18n/emqx_bridge_api.conf [hunk 2]
```diff
@@ -134,4 +134,20 @@ NOTE：不允许在单节点上启用/禁用 Bridge"""
...
+The ID must be of format '{type}:{name}'
+"""
+                       zh: """[FIXME]
+通过给定的 ID 测试创建一个新的连接器 </br>
+ID 的格式必须为“{type}:{name}”
+"""
+                      }
+                 label: {
+                        en: "Create Test Bridge"
+                        zh: "[FIXME] 创建测试连接器"
```
Comment (zmstone) [inline apps/emqx_bridge/i18n/emqx_bridge_api.conf:150]:
```suggestion
                        en: "Test Bridge Creation"
                        zh: "测试桥接创建"
```

File: apps/emqx_bridge/i18n/emqx_bridge_api.conf [hunk 3]
```diff
@@ -134,4 +134,20 @@ NOTE：不允许在单节点上启用/禁用 Bridge"""
...
 
+    desc_api9 {
+                 desc {
+                       en: """
+Test creating a new bridge by given ID </br>
+The ID must be of format '{type}:{name}'
+"""
+                       zh: """[FIXME]
+通过给定的 ID 测试创建一个新的连接器 </br>
+ID 的格式必须为“{type}:{name}”
```
Comment (zmstone) [inline apps/emqx_bridge/i18n/emqx_bridge_api.conf:145]:
```suggestion
                       zh: """通过给定的 ID 测试创建一个新的桥接。 </br>
ID 的格式必须为 ’{type}:{name}‘。
```

**Review by zmstone — APPROVED** [review]

**Comment by zmstone** [comment]
as discussed: should follow up with PR to refactor
* make resource creating dry_run timeout configurable (currently 15 seconds)
* remove the gen_tcp:connect probing from outside the resource implementation

File: apps/emqx_bridge/src/emqx_bridge_api.erl [hunk 4]
```diff
@@ -462,6 +481,59 @@ schema("/nodes/:node/bridges/:id/operation/:operation") ->
...
+    {ok, #{host := Host, port := Port}} = emqx_http_lib:uri_parse(BaseUrl),
+    {Host, Port};
+host_and_port(_Unknown, _) ->
+    undefined.
+
+test_connection(undefined) ->
+    %% be friendly, it might fail later on with a 'timeout' error.
+    ok;
+test_connection({Host, Port}) ->
+    case gen_tcp:connect(Host, Port, []) of
```
Comment (zmstone) [inline apps/emqx_bridge/src/emqx_bridge_api.erl:519]: there should be a timeout here.

**Comment by sstrigler** [comment]
Follow up ticket: https://emqx.atlassian.net/browse/EMQX-8588

File: apps/emqx_bridge/src/emqx_bridge_api.erl [hunk 5]
```diff
@@ -462,6 +481,59 @@ schema("/nodes/:node/bridges/:id/operation/:operation") ->
...
+        {ok, TestSocket} -> gen_tcp:close(TestSocket);
+        Error -> Error
+    end.
+
+parse_url(Url) ->
+    case string:split(Url, "//", leading) of
```
[the remark this answers is not in the document]
Comment (sstrigler) [reply apps/emqx_bridge/src/emqx_bridge_api.erl:526]: next time, going to be deleted anyway :)

**Pull request merged**